pkg/abstractions/image/build.go

```diff
@@ -512,3 +515,75 @@ func hasAnyPrefix(s string, prefixes []string) bool {
 	}
 	return false
 }
+
+func parseFlagLinesAndPackages(pythonPackages []string) ([]string, []string) {
+	var flagLines []string
+	var packages []string
+	var flags = []string{"--", "-"}
+
+	for _, pkg := range pythonPackages {
+		if hasAnyPrefix(pkg, flags) {
+			flagLines = append(flagLines, pkg)
+		} else {
+			packages = append(packages, fmt.Sprintf("%q", pkg))
+		}
+	}
+	return flagLines, packages
+}
+
+// Generate the commands to run in the container. This function will coalesce pip and mamba commands
+// into a single command if they are adjacent to each other.
+func generateCommands(buildSteps []BuildStep, pythonVersion string) []string {
```

Comment:
Can we add a test for this? Want to ensure we're grouping adjacent commands and not all of the commands so that we can ensure the ordering is still intact.

Reply:
Yeah I'll write up some tests